client/client-lib/src/lib.rs

```diff
@@ -263,23 +263,24 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
     }
 
     /// Fetches the client secret from the database or generates a new one if none is present
-    fn get_secret(db: &Database) -> DerivableSecret {
+    async fn get_secret(db: &Database) -> DerivableSecret {
         let mut tx = db.begin_transaction(ModuleRegistry::default());
-        let secret = tx
-            .get_value(&ClientSecretKey)
-            .expect("DB error")
-            .unwrap_or_else(|| {
-                let secret: ClientSecret = thread_rng().gen();
-                let no_replacement = tx
-                    .insert_entry(&ClientSecretKey, &secret)
-                    .expect("DB error")
-                    .is_none();
-                assert!(
-                    no_replacement,
-                    "We would have overwritten our secret key, aborting!"
-                );
-                secret
-            });
+        let client_secret = tx.get_value(&ClientSecretKey).expect("DB error");
+        let secret = if let Some(client_secret) = client_secret {
```

Comment:
let me know if there's a more "rust-like" way of doing this

Reply:
That looks like a case for let-else. Does our rust version in `nix develop` support that yet? I though we upgraded a while ago @dpc?